quodlibet/quodlibet/ext/events/bansheeimport.py

```diff
@@ -0,0 +1,176 @@
+
+            # rating is stored as integer from 0 to 5
+            b_rating = row["Rating"] / 5.0
+            if b_rating != song("~#rating"):
+                song["~#rating"] = b_rating
+                has_changed = True
+
+            # play count is stored as integer from 0
+            if row["PlayCount"] != song("~#playcount"):
+                # summing play counts would break on multiple imports
+                song["~#playcount"] = row["PlayCount"]
+                has_changed = True
+
+            # skip count is stored as integer from 0
+            if row["SkipCount"] != song("~#skipcount"):
+                song["~#skipcount"] = row["SkipCount"]
+                has_changed = True
+
+            # timestamp is stored as integer or None
+            if row["LastPlayedStamp"] is not None:
+                value = row["LastPlayedStamp"]
+                # keep timestamp if it is newer than what we had
+                if value > song("~#lastplayed", 0):
+                    song["~#lastplayed"] = value
+                    has_changed = True
+
+            if row["DateAddedStamp"] is not None:
+                value = row["DateAddedStamp"]
+                # keep timestamp if it is older than what we had
+                if value < song("~#added", 0):
+                    song["~#added"] = value
+                    has_changed = True
+
+            if has_changed:
+                self._changed_songs.append(song)
+
+    def finish(self):
+        """Call at the end, also returns number of songs with data imported"""
+
+        count = len(self._changed_songs)
+        self._library.changed(self._changed_songs)
+        self._changed_songs = []
+        return count
+
+
+def do_import(parent, library):
+    db_path = expanduser(BansheeImport.USR_PATH)
+    db = sqlite3.connect(db_path)
+
+    importer = BansheeDBImporter(library)
+    try:
+        importer.read(db)
+    except sqlite3.OperationalError:
+        msg = _("Specified Banshee database is malformed or missing")
+        WarningMessage(parent, BansheeImport.PLUGIN_NAME, msg).run()
+    except Exception:
+        util.print_exc()
+        importer.finish()
+        msg = _("Import Failed")
+        # FIXME: don't depend on the plugin class here
+        ErrorMessage(parent, BansheeImport.PLUGIN_NAME, msg).run()
+    else:
+        count = importer.finish()
+        preamble = "Successfully imported ratings and statistics for "
+        msg = ngettext(preamble + "%d song", preamble + "%d songs",
+                       count) % count
+        Message(Gtk.MessageType.INFO, parent, BansheeImport.PLUGIN_NAME,
+                msg).run()
+
+    db.close()
+
+
+class BansheeImport(EventPlugin):
+    PLUGIN_ID = "bansheeimport"
+    PLUGIN_NAME = _("Banshee Import")
+    PLUGIN_DESC = _("Imports ratings and song statistics from Banshee.")
+    PLUGIN_ICON = Icons.DOCUMENT_OPEN
+
+    DEF_PATH = "~/.config/banshee-1/banshee.db"
+    USR_PATH = DEF_PATH
```

Comment:
Why would the DB be in another place?

Reply:
The directory depends on the XDG settings of the user, though it defaults to here. We could instead read the XDG settings and infer the location that way, but perhaps it could have changed. I don't see a reason *not* to leave this as an option to the user